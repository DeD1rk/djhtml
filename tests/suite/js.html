<!-- Basic Javascript -->
<script>
    function f() {
        return 42;
    }
</script>


<!-- Different brace types -->
<script>
    array = [
        1,
        2,
        3,
    ]

    dictionary = {
        value1: 1,
        value2: 2,
        value3: 3,
    }

    function_call(
        arg1,
        arg2,
        arg3,
    )
</script>


<!-- Function chaining -->
<script>
    one()
        .two()
        .three();
    four();
</script>

<script>
    function f() {
        return 42;
    }
</script>


<!-- Double-nested braces -->
<script>
    $(function() {
        this.addEventListener('change', function () {
            doStuff();
        });
    });
</script>


<!-- Multiline comment -->
<script>
    /*
     * Hi mom!
     *
     */
</script>


<!-- Nasty characters embedded in strings -->
<script>
    console.log("$(function() {");
    console.log('$(function() {');
    console.log(`$(function() {`);
    console.log(`
function f() {
return 42;
}
    `);
</script>


<!-- Real-life example kindly provided by @jacklinke -->
<script>
    const format_duration_value = (a) => {
        let secs = a * 3600;  // get total amount of seconds from your input
        const days = Math.floor(secs / (3600 * 24));
        return (`
            <div class="form-inline text-nowrap text-center">
                ${days.toString().padStart(2,'0')}
                <small>m</small>
            </div>
        `);
    };
</script>


<!-- More Javascript shenanigans -->
<script>
    $(function1()
        {
            var foo = (function2(){
                doStuff();
            })();
        });
    // Jeez
</script>


<!-- If elif else -->
<script>
    let time = 10;
    if (time < 10) {
        console.log("Good morning");
    } else if (time < 20) {
        console.log("Good day");
    } else {
        console.log("Good evening");
    }
</script>


<!-- Switch case indentation -->
<script>
    const mode = 'approved'
    switch (mode) {
        case 'approved':
            console.log('Approved');
            break;
        case 'rejected':
            console.log('Rejected');
            break;
        default:
            console.log('Default');
    }

    switch (mode) {
        case 'approved':
            console.log('Approved');
            break;
        default:
            console.log('Default');
        case 'rejected':
            console.log('Rejected');
            break;
    }

    switch (mode) {
        case 'approved':
        case 'rejected':
        case 'invalid':
            console.log('Not default');
            break;
        default:
            console.log('Default');
    }

    switch (mode) {
        case 'approved': {
            console.log('Approved');
            break;
        }
        case 'rejected': {
            console.log('Rejected');
            break;
        }
        default: {
            console.log('Default');
        }
    }
</script>


<!-- Switch complex example -->
<script>
    var year = 2016;
    var month = 2;
    var dayCount;
    switch (month) {
        case 1:
            function switchFoo () {
                console.log("Foo");
            }
            switchFoo();
        case 12:
            dayCount = 31;
            break;
        case 4:
            switch(boo){
                case 1:
                case 2:
            }
        case 11:
            dayCount = 30;
            break;
        case 2:
            if (((year % 4 == 0) && !(year % 100 == 0)) || (year % 400 == 0)) {
                dayCount = 29;
            } else {
                dayCount = 28;
            }
            break;
        default:
            dayCount = -1; // invalid month
    }
    console.log(dayCount);
</script>


<!-- Nested switch statements -->
<script>
    switch (a) {
        case b:
            switch (c) {
                case d:
                    switch (e) {
                        default:
                    }
            }
    }
</script>


<!-- Things that look like switch, but aren't -->
<script>
    function switchBoo () {
        console.log("Boo");
    }
    switchBoo();

    let arr = [
        'switch',
        'default:',
        'case ',
    ];
</script>


<!-- Another problematic switch statement -->
<script>
    switch (foo) {
        case bar:
            if (baz) {
                switch (foo) {
                    case bar:
                        if (baz) {
                            xizzy();
                        }
                }
            }
    }
</script>

<!-- More function chaining -->
<script>
    window.fetch('/test.html')
        .then((html) => {
            document.body.innerHTML = html;
        });

    foo
        .bar
        .baz
        .then(() => {
            foo
                .bar
                .baz
                .then(() => {
                    foo
                        .bar
                        .baz
                });
        });
</script>


<!-- Not enough closing tokens -->
<head>
    <script>
        function foo() {
    </script>
</head>

<!-- Too much closing tokens -->
<head>
    <script>
        function foo() {
        }
        }
        }
    </script>
</head>
