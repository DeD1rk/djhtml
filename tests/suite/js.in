<!-- Basic Javascript -->
<script>
function f() {
return 42;
}
</script>


<!-- Different brace types -->
<script>
array = [
1,
2,
3,
]

dictionary = {
value1: 1,
value2: 2,
value3: 3,
}

function_call(
arg1,
arg2,
arg3,
)
</script>


<!-- Function chaining -->
<script>
one()
.two()
.three();
four();
</script>

<script>
function f() {
return 42;
}
</script>


<!-- Double-nested braces -->
<script>
$(function() {
this.addEventListener('change', function () {
doStuff();
});
});
</script>


<!-- Multiline comment -->
<script>
    /*
     * Hi mom!
     *
     */
</script>
