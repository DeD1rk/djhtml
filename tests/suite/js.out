<!-- Basic Javascript -->
<script>
    function f() {
        return 42;
    }
</script>


<!-- Different brace types -->
<script>
    array = [
        1,
        2,
        3,
    ]

    dictionary = {
        value1: 1,
        value2: 2,
        value3: 3,
    }

    function_call(
        arg1,
        arg2,
        arg3,
    )
</script>


<!-- Function chaining -->
<script>
    one()
        .two()
        .three();
    four();
</script>

<script>
    function f() {
        return 42;
    }
</script>


<!-- Double-nested braces -->
<script>
    $(function() {
        this.addEventListener('change', function () {
            doStuff();
        });
    });
</script>


<!-- Multiline comment -->
<script>
    /*
     * Hi mom!
     *
     */
</script>


<!-- Nasty characters embedded in strings -->
<script>
    console.log("$(function() {");
    console.log('$(function() {');
    console.log(`$(function() {`);
    console.log(`
function f() {
return 42;
}
    `);
</script>


<!-- Real-life example kindly provided by @jacklinke -->
<script>
    const format_duration_value = (a) => {
        let secs = a * 3600;  // get total amount of seconds from your input
        const days = Math.floor(secs / (3600 * 24));
        return (`
            <div class="form-inline text-nowrap text-center">
                ${days.toString().padStart(2,'0')}
                <small>m</small>
            </div>
        `);
    };
</script>


<!-- More Javascript shenanigans -->
<script>
    $(function1()
        {
            var foo = (function2(){
                doStuff();
            })();
        });
    // Jeez
</script>

